File: spring-security-in-action/ssia-ch07-ex3/src/test/java/org/example/ssiach07ex3/controller/HelloControllerTest.java
package org.example.ssiach07ex3.controller;

import static org.junit.jupiter.api.Assertions.*;
import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.*;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
import org.springframework.test.web.servlet.result.MockMvcResultMatchers;

@SpringBootTest
@AutoConfigureMockMvc
class HelloControllerTest {

    @Autowired
    private MockMvc mvc;

    @Test
    void testHello() throws Exception {
        mvc.perform(get("/hello").with(httpBasic("john", "12345")))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$").value("Hello!"));
    }

    @Test
    void testHello2() throws Exception {
        mvc.perform(get("/hello").with(httpBasic("jane", "12345")))
                .andExpect(status().isForbidden());
    }
}

File: spring-security-in-action/ssia-ch2-ex2/src/test/kotlin/org/example/ssiach2ex2/controller/HelloControllerTest.kt
package org.example.ssiach2ex2.controller

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*

@SpringBootTest
@AutoConfigureMockMvc
class HelloControllerTest {

    @Autowired
    private lateinit var mockMvc: MockMvc

    @Test
    fun testHello() {
        mockMvc.perform(MockMvcRequestBuilders.get("/hello")
            .with(SecurityMockMvcRequestPostProcessors.httpBasic("john", "12345")))
            .andExpect(status().isOk)
            .andExpect(jsonPath("$").value("Hello World!"))
    }

    @Test
    fun testFailHello() {
        mockMvc.perform(MockMvcRequestBuilders.get("/hello"))
            .andExpect(status().is4xxClientError)
    }
}